edb/server/compiler/compiler.py

```diff
@@ -684,9 +760,13 @@ def _compile_ql_query(
                 ir.schema, params_type = s_types.Tuple.create(
                     ir.schema, element_types={}, named=False)
 
-            in_type_data, in_type_id = sertypes.TypeSerializer.describe(
-                ir.schema, params_type, {}, {},
-                protocol_version=ctx.protocol_version)
+            if ctx.protocol_version >= (0, 12):
```

Comment:
I _think_ this the last bug. Which branch should be taken when the protocol is `None` (HTTP)? This currently fails with `TypeError: '>=' not supported between instances of 'NoneType' and 'tuple'`

Reply:
The user-facing HTTP interface is not changing, so the HTTP extension should always use the latest version (and set it explicitly).